### ISSUE TITLE
[a11y] Accessibility for nested text components

### ISSUE THREAD

**Issue by frags51** [facebook/react-native#32004]
## Description
Same as (stale) https://github.com/facebook/react-native/issues/27147. 
Consider this example:
```
<Text>
    <Text> Please click:  </Text>
    <Text accessible={true} accessibilityRole='link' onPress={openLink()}> Link </Text>
</Text>
```

A screen reader user is not able to click/open the link (i.e. the 2nd nested Text component). Ideally, the screen reader should focus on the link as well. The addition of any permutation of accessibility props on the parent/child Text component does not help.

## React Native version:
**React native 0.62.**
React Native Environment Info:
    System:
      OS: Windows 10
      CPU: (12) x64 AMD Ryzen 5 5600X 6-Core Processor
      Memory: 19.72 GB / 31.92 GB
    Binaries:
      Node: 12.21.0 - C:\Users\supsing\AppData\Local\Temp\yarn--1628676998345-0.3639484914346074\node.CMD
      Yarn: 1.22.10 - C:\Users\supsing\AppData\Local\Temp\yarn--1628676998345-0.3639484914346074\yarn.CMD
      npm: 6.14.11 - C:\Users\supsing\AppData\Local\nvs\default\npm.CMD
      Watchman: 20210102.202219.0 - E:\watchman-v2021.01.04.00-windows\bin\watchman.EXE

## Steps To Reproduce
Write nested text components as shown in the description.

## Expected Results
Screen reader (Voiceover/talkback) should be able to tell the user that there is a link and user should be able to click on that link.

In Android (native), talkback adds any links available in a single text view in the accessibility menu (can be opened using swipe up then swipe right). The same should be done by React Native as well. Perhaps it can be achieved with ClickableSpan in Android. There should be similar ways in iOS as well.
## Snack, code example, screenshot, or link to a repository:
```
<Text>
    <Text> Please click:  </Text>
    <Text accessible={true} accessibilityRole='link' onPress={openLink()}> Link </Text>
</Text>
```

**Comment by frags51** [comment]
There is no easy solution for this, even if try to create empty, pressable Views just where the pressable link is. This is because, onTextLayout/onLayout does not fire for the nested Text elements and hence it is not possible to know their position.

**Comment by jsamr** [comment]
@amarlette, that's a big issue for consumers of our library https://github.com/meliorence/react-native-render-html, do you think this issue [could be addressed in the context of the GAAD Pledge](https://reactnative.dev/blog/2021/03/08/GAAD-React-Native-Accessibility)?

**Comment by tj-mc** [comment]
@jsamr Do you have any related issue threads from `react-native-render-html` that might help us understanding how this issue is affecting users?

**Comment by jsamr** [comment]
@tj-mc  At least one, yes! https://github.com/meliorence/react-native-render-html/issues/285
The issue was also raised in our Discord server,  [such as here](https://discord.com/channels/736906960041148476/736906960041148480/901834583190954025).

**Comment by tj-mc** [comment]
Thanks. I'm doing my best to understand what technical solution would be required here, and if it's within my reach, as I also feel this is an important issue. 

I agree @jsamr that this is a duplicate of #31298. @frags51, Would we consider closing this issue, and continuing this conversation in #31298?

**Comment by fabriziobertoglio1987** [comment]
>In Android (native), talkback adds any links available in a single text view in the accessibility menu (can be opened using swipe up then swipe right). The same should be done by React Native as well.

as documented in my test case https://github.com/fabriziobertoglio1987/react-native-notes/issues/9#issuecomment-1045593386 this was solved with commit https://github.com/facebook/react-native/commit/b352e2da8137452f66717cf1cecb2e72abd727d7 on Android

>A screen reader user is not able to click/open the link (i.e. the 2nd nested Text component). Ideally, the screen reader should focus on the link as well. 

currently working on https://github.com/fabriziobertoglio1987/react-native-notes/issues/9#issuecomment-1045593164. thanks

**Comment by blavalla** [comment]
@frags51 , has this been resolved by @fabriziobertoglio1987 's PR?

**Comment by frags51** [comment]
@fabriziobertoglio1987 Is this working on iOS as well? @blavalla if not, then it needs to be solved for iOS too.

**Comment by tanyalsweeney** [comment]
Is there at least an iOS workaround? I'm experiencing this now

**Comment by pranjal-jately-unmind** [comment]
Thanks for addressing this on Android, @fabriziobertoglio1987. This is not working on iOS currently, and it would be great to have a fix for that as well, thanks.

**Comment by volpstar** [comment]
Why wouldn't you just wrap the text in a Touchable? Semantically, this is more accurate I think.

**Comment by fabriziobertoglio1987** [comment]
iOS API ([RCTParagraphComponentAccessibilityProvider](https://github.com/facebook/react-native/blob/29a13b69899744c889fddb5e969808186f8db669/React/Fabric/Mounting/ComponentViews/Text/RCTParagraphComponentAccessibilityProvider.mm#L48-L131), related PR [#35432](https://github.com/facebook/react-native/pull/35432)).

**Comment by Drew-Gerber** [comment]
We're considering using this approach for handling inline links on iOS: https://benjie.ca/blog/2021/09/21/accessible-inline-links-react-native.html

It's not perfect, but at least makes the links available to VoiceOver users. An alternative approach might be to use an accessibility action.

**Comment by stevehanson** [comment]
Has anyone found a solution for this that works for more than one link in a paragraph? I am building a news app for a major publication. We must meet WCAG AA, which I believe all apps should aim for, and I am struggling to find a path forward. This issue has been literally keeping me up at night.

We don't have the time or resources at this stage in the project to upgrade to the new architecture, which seems risky anyways, given that Expo only just now released experimental support (we're on Expo), and a quick spike I did that uncovered some bugs we'd have to resolve.

I'm shocked that React Native has such a fundamental accessibility gap here and that there hasn't been an effort to port the fix to the old architecture.

**Comment by tj-mc** [comment]
I agree this is an unfortunate gap. Is it solved in the new architecture? 

Perhaps if you have rich text content, an inline webview might be a better solution. You can load local HTML and render without a loading time if it's static.

It seems like most news apps I use are native wrappers around web layouts.

**Comment by stevehanson** [comment]
Thanks for this suggestion, @tj-mc! I converted this portion of my app to use webviews instead, and this completely solved my accessibility issues. Screen reader, voice control, keyboard all work exactly as expected. It's a shame that we have to use webviews as an escape hatch, and I certainly wouldn't consider this issue as solved because of that option, but I am glad to have found a way to move forward.

**Comment by frags51** [comment]
@stevehanson The changes shown in [this PR](https://github.com/BigThinkcode/AdaptiveCards/pull/6/files#diff-bb8d71b774dd05a3fa0f1f3186f77c8e5f90eb081c39fdbf60630f498c89bbc8) could be a workaround.

Webview/HTML definitely helps but do consider the perf/memory implications of using one.

**Comment by chriszs** [comment]
After testing with Accessibility Inspector on iOS Simulator and TalkBack on the Android emulator, my conclusion is that this does appear to be adequately solved on the new architecture on iOS and with TalkBack on the old renderer, but is not solved on the old renderer on iOS, as also noted in #35193. VoiceOver reads the content of the link, but it is not possible to independently select it with VoiceOver or open the link. I realize this is just a summation of what many have concluded above, but I was confused for a minute reading through this issue and wanted to confirm the scope.

**Comment by msftedad** [comment]
@chriszs,  The role of link must be announced to user as if it does not announce the role user who depends on Voiceover will be impacted , Also Voice over is announcing it as Static text with no other information like double tap to activate the link.

**Comment by chriszs** [comment]
@msftedad Good point. Do you think that works okay on the new architecture and Android (where Android offers to provide a list of links after the paragraph has been read)?

**Comment by msftedad** [comment]
Hi @chriszs, This was discussed and as per the Inputs, this will be read on Android as well, so user will be aware of the information in paragraph.

**Comment by msftedad** [comment]
@chriszs , Any update?

**Comment by yanachrnsh** [comment]
I faced the same issue, if I put Text inside Text with accessibility role "link" and onPress, on the Android emulator it says that link can be opened using swipe up then swipe right, but on iOS it does not recognise  as it is link

```
<Text> 
   <Fragment>Any text </Fragment>
    <Text accessibilityRole='link' onPress={ ()=>{}}>Link</Text>
</Text>
```

**Comment by msftedad** [comment]
Issue is still repro'ing.Please don't close this bug.

**Comment by dkaleBB** [comment]
Hello,
I recently encountered this issue as well where VO would reach the anchor text but not allow me to double tap on it. This also happens when my Text in nexted. Would really appreciate a fix

**Comment by alexandrermo** [comment]
I am experiencing this bug at 0.79.6, testes on Android 16, api 36.

**Comment by yanachrnsh** [comment]
Please do not close this ticket, still expecting the same issue. 
A few updates after testing it on different environments/devices:  


Device/Platform | Result
-- | --
iPhone 15 (iOS,  Simulator, Expo Go) | ✅ Works as expected
Android 13 (Galaxy A32, Physical Device, Expo Go) | ✅ Works as expected
Android A35 (BrowserStack, Dev Build) | ✅ Works as expected
iPhone 13 (iOS, BrowserStack, Dev Build) | ❌ Does not work
iPhone 15 (iOS, Simulator, Dev Build) | ❌ Does not work
Android 13 (Galaxy A32, Physical Device, Dev Build) | ✅ Works as expected

**Comment by bharath2468** [comment]
@yanachrnsh , Do you able to check for other a11y apps like select to speak, talkback, switch access and voice access?. I see the Text is read twice in select to speak and unable to access linkText via keyboard.

**Comment by msftedad** [comment]
Hi Team, Any update on this issue?
